### PR TITLE
Question Display with Pagination

### LINKED ISSUE #8: REQ-007: Paginated Question Display
# Paginated Question Display Implementation

## Description
Display questions in groups of 5, showing complete question information including options and correct answer. Implement a pagination system for navigation.

### Display Format
```
===============================
QUESTION LIST (Page 1 of X)
===============================
[Question 1]
ID: 1
Q: What is...?
1. Option A
2. Option B
3. Option C
Correct Answer: 2

[Question 2]
ID: 2
Q: Second question...?
1. Option X
2. Option Y
3. Option Z
Correct Answer: 1

[...]
===============================
1. Next Page
2. Exit to Menu
Option: _
```

### Core Features
* Display questions in groups of 5
* Each question display must include:
  - Question ID
  - Question text
  - All three options
  - Correct answer
* Navigation system:
  - Next page option
  - Exit to menu option
* Page information display
* Clear formatting

## Acceptance Criteria
- [ ] Displays exactly 5 questions per page
- [ ] Shows all required question information
- [ ] Navigation works properly:
  - Next page works
  - Exit to menu works
- [ ] Page counting is correct
- [ ] Handles last page properly
- [ ] Proper formatting and readability
- [ ] Handles edge cases:
  - Empty list
  - Less than 5 questions
  - Last page with fewer questions

## Technical Notes
* Implement page calculation
* Clear screen management
* Input validation
* Error handling:
  - No questions
  - Invalid input
  - Navigation boundaries
* Memory efficiency

## Integration Points
* Works with Question structure
* Uses existing question access functions
* Integrates with main menu system

### PR DESCRIPTION
**Pull request by Starfisheunyoung**
**Description** 
This Pull Request introduces the functionality for displaying questions in pages, where each page contains a set of 5 questions

**Features Implemented:**

1. **Display Questions in Pages**:

   - Questions are displayed in pages, with a maximum of 5 questions per page.

   - The user can navigate between pages using the options 1. for Next page, and 2. for exiting to the menu. 

   - Each question on the page shows the question text, options, and the correct answer.

2. **Pagination System**:

   - The pagination system dynamically calculates the total number of pages based on the number of questions and the number of questions to be shown per page.

   - Proper handling of the navigation between pages ensures that the user can scroll through all the questions without going out of bounds.

3. **Improved User Interaction**:

   - A clear and user-friendly interface is provided with options to navigate through the pages of questions or return to the main menu.

## Tasks Completed:

- Implemented the pagination system to display 5 questions per page.

- Displayed question details (text, options, correct answer) in a user-friendly format.

-Added functionality for navigating between pages.

-Implemented "Exit to Menu" option for easy return to the main menu.

### CODE REVIEW

File: marathon_quiz.cpp [hunk 1]
```diff
@@ -93,6 +93,9 @@ float newScore(int,int);
 //Nickname creation function prototype definition
 void nicknameCreation(char*);
 
+//Funciton for display quetions page
+void displayQuestionsInPages(Question* questionHead)
```
Comment (Friedamariao) [inline marathon_quiz.cpp:97]: @Starfisheunyoung Please add the missing semicolon at the end of the function prototype displayQuestionsInPages

File: marathon_quiz.cpp [hunk 2]
```diff
@@ -143,13 +146,15 @@ int main(){
             	displayQuestionsByCategory(questionHead, category);
             	break;
             case 8:
+            	displayQuestionsInPages(Question* questionHead);
```
Comment (Friedamariao) [inline marathon_quiz.cpp:149]: @Starfisheunyoung 
- In line 149, let's leave out the type declaration inside of the function call, it's only needes the quesionHead. 
- Please also add, the break sentence at the end of case 8.

File: marathon_quiz.cpp (hunk 1 above)
Comment (Friedamariao) [inline marathon_quiz.cpp]: @Starfisheunyoung Please add the missing semicolon at the end of the displayQuestionsInPages function declaration.

File: marathon_quiz.cpp (hunk 2 above)
Comment (Friedamariao) [inline marathon_quiz.cpp]: @Starfisheunyoung 
- In line 149, let's remove the type declaration inside the function call; it only needs `questionHead` as the argument.
- Please add the missing `break;` sentence at the end of case 8.

**Review by Friedamariao — CHANGES_REQUESTED** [review]

File: marathon_quiz.cpp [hunk 3]
```diff
@@ -415,6 +421,73 @@ void displayQuestionsByCategory(Question* head, int category){
 	printf("\n");
 }
 /*
+Display questions page
+*/
+void displayQuestionsInPages(Question* questionHead) {
```
Comment (Friedamariao) [inline marathon_quiz.cpp:522]: @Starfisheunyoung
I think there might be some issues with the logic of the function. After making the necessary changes to get the program to compile correctly, here are the problems I noticed:

- The functionality allows moving to the next page, but it doesn't work when trying to go back to the previous page; it just says "try again."
- The first question on the second page is numbered as '1' again, but it should continue from the previous page, showing 6, 7, 8, and so on.
- Lastly, the page numbers seem to be off. On the first page, it correctly displays "Page 1 of 2," but on the second page, it shows "Page 1 of 1."

You might want to check the `printPlayers` function, which also has a pagination system that works like this one should work!

**Review by Friedamariao — CHANGES_REQUESTED** [review]

**Review by gfdfranco — APPROVED** [review]
LGTM

**Review by DanielRV3 — APPROVED** [review]
LGTM

**Review by Friedamariao — APPROVED** [review]

**Pull request merged**